### PR TITLE
ICMSLST-951 Optimise commodity queries and update Sanctions with commodity unit annotation.

### PR DESCRIPTION
**Pull request by MattHolmes123**

### CODE REVIEW

File: web/utils/commodity.py [hunk 1]
```diff
@@ -62,10 +84,22 @@ def get_usage_commodities(application_usage: "QuerySet[Usage]") -> "QuerySet[Com
...
+    :param model: A model linked to commodities
+    :param commodity_path: The path indicating the relationship to the commodity model.
+    :return: annotated model
+    """
+
+    commodity_unit = Unit.objects.filter(
+        pk=OuterRef(f"{commodity_path}commoditygroup__unit")
+    ).order_by("-pk")
+
+    return model.annotate(unit_descriptiion=Subquery(commodity_unit.values("description")[:1]))
```
Comment (kevinetienne) [inline web/utils/commodity.py:102]: Small typo here: `unit_descriptiion` -> `unit_description`

File: web/utils/commodity.py [hunk 2]
```diff
@@ -62,10 +84,22 @@ def get_usage_commodities(application_usage: "QuerySet[Usage]") -> "QuerySet[Com
...
     groups = get_usage_commodity_groups(application_usage)
 
-    return Commodity.objects.filter(
-        commoditygroup__in=groups,
-        is_active=True,
-    ).distinct()
+    return Commodity.objects.filter(commoditygroup__in=groups, is_active=True).distinct()
+
+
+def annotate_commodity_unit(model: "QuerySet[Model]", commodity_path=""):
```
Comment (kevinetienne) [inline web/utils/commodity.py:90]: do you need yet to make `commodity_path` dynamic? It looks like it's `commodity__` in this PR at least but maybe you've got plan already for it?

File: web/templates/web/domains/case/import/sanctions/view.html [hunk 3]
```diff
@@ -33,7 +33,7 @@
             <td>{{ commodity_link(row.commodity) }}</td>
             <td>{{ row.goods_description }}</td>
             <td>{{ row.quantity_amount }}</td>
-            <td>kilos</td>
+            <td>{{ row.unit_descriptiion }}</td>
```
Comment (kevinetienne) [inline web/templates/web/domains/case/import/sanctions/view.html:36]: Small typo here: `unit_descriptiion` -> `unit_description`

File: web/templates/web/domains/case/import/sanctions/edit_application.html [hunk 4]
```diff
@@ -48,7 +48,7 @@ <h4>Goods</h4>
                         <td>{{ row.commodity }}</td>
                         <td>{{ row.goods_description }}</td>
                         <td>{{ row.quantity_amount }}</td>
-                        <td>kilos</td>
+                        <td>{{ row.unit_descriptiion }}</td>
```
Comment (kevinetienne) [inline web/templates/web/domains/case/import/sanctions/edit_application.html:51]: Small typo here: `unit_descriptiion` -> `unit_description`

File: web/utils/commodity.py (hunk 1 above)
Comment (kevinetienne) [reply web/utils/commodity.py:102]: I know we discussed it and we might not have enough data but did we have cases where we had different units?

Comment (mikeylockwood) [reply web/utils/commodity.py:102]: Is this query to annotate `unit_description` with the unit description from the first commodity unit?

Maybe something like
```
    commodity_unit = Unit.objects.filter(
        pk=OuterRef(f"{commodity_path}commoditygroup__unit")
    ).order_by("-pk").first()
    
    model.annotate(unit_description=Value(commodity_unit.description, output_field=CharField()))
  ```

File: web/utils/commodity.py (hunk 2 above)
Comment (MattHolmes123) [reply web/utils/commodity.py:90]: I didn't want to bake in the link to commodity as it could be different

File: web/utils/commodity.py (hunk 1 above)
Comment (MattHolmes123) [reply web/utils/commodity.py:102]: As discussed it just picks the latest group unit.

**Review by kevinetienne — APPROVED** [review]

**Review by mikeylockwood — APPROVED** [review]

**Pull request merged**